### PR TITLE
[GUVNOR-3377] - Implement tests for https support in Kie Server

### PR DESCRIPTION
**Pull request by jiripetrlik**

### CODE REVIEW

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/HttpsUtils.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        final SSLConnectionSocketFactory sslsf = getSSLConnectionSocketFactory();
+        return HttpClients.custom().setSSLSocketFactory(sslsf).build();
+    }
+
+    public static CloseableHttpClient createHttpClient(CredentialsProvider credentialsProvider) {
+        final SSLConnectionSocketFactory sslsf = getSSLConnectionSocketFactory();
+        return HttpClients.custom().setSSLSocketFactory(sslsf).setDefaultCredentialsProvider(credentialsProvider).build();
+    }
+
+    public static String readResonseContent(CloseableHttpResponse response) {
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/HttpsUtils.java]: readResonseContent -> readResponseContent

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/HttpsUtils.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package org.kie.cloud.integrationtests.https;
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/HttpsUtils.java]: Missing licence header

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/KieServerHttpsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
+package org.kie.cloud.integrationtests.https;
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/KieServerHttpsIntegrationTest.java]: Missing licence header

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/KieServerHttpsIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                deploymentScenario.getKieServerDeployment().getPassword());
+        try (CloseableHttpClient httpClient = HttpsUtils.createHttpClient(credentialsProvider)) {
+            try (CloseableHttpResponse response = httpClient.execute(serverInforRequest())) {
+                Assertions.assertThat(response.getStatusLine().getStatusCode()).isEqualTo(HttpsURLConnection.HTTP_OK);
+
+                final String responseContent = HttpsUtils.readResonseContent(response);
+                Assertions.assertThat(responseContent).contains(KIE_SERVER_INFO_TEXT);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to connect to KIE server REST API", e);
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/KieServerHttpsIntegrationTest.java]: You can avoid these catch statements by "throws Exception" on method level.

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/HttpsUtils.java (hunk 2 above)
Comment (jiripetrlik) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/HttpsUtils.java]: Fixed in next commit.

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/KieServerHttpsIntegrationTest.java (hunk 3 above)
Comment (jiripetrlik) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/KieServerHttpsIntegrationTest.java]: Fixed in next commit.

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/KieServerHttpsIntegrationTest.java (hunk 4 above)
Comment (jiripetrlik) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/KieServerHttpsIntegrationTest.java]: Yes, you are right. However, I still have to catch IOException.

**Comment by jiripetrlik** [comment]
Fixed according to review.

**Pull request merged**